spec2nii/GE/ge_read_pfile.py

```diff
@@ -1290,27 +1300,33 @@ def read_data(self):
             X1, X2 = np.meshgrid(np.arange(dataframes), np.arange(nechoes))
             X1 = X1.T.ravel()
             X2 = X2.T.ravel()
-            Y1 = (-1)**(noadd * X1)
+            # Do not apply any phase cycling correction if the receiver phase toggle in sLASER was set
+            if cv24 >= 16384:
+                Y1 = np.ones_like(X1)
+            else:
+                Y1 = (-1) ** (noadd * X1)
             Y1 = np.moveaxis(np.broadcast_to(Y1, (1, 1, 1, npoints, nreceivers, Y1.size)), -1, -2)
             Y2 = (1 + refframes + (totalframes / nechoes) * X2 + X1).astype(int)
 
             self.raw_suppressed = self.raw_data[:, :, :, :, Y2, :] * Y1 * mult
 
-            # Up to this point we have simply replicated the logic of the GELoad function.
-            # Now reorganise dimensions to give a editing dimension.
-            # This means that this is done in a not particularly clear order, but it enables testing against
-            # the matlab code.
+        # Up to this point we have simply replicated the logic of the GELoad function.
+        # Now reorganise dimensions to give an editing dimension.
+        # This means that this is done in a not particularly clear order, but it enables testing against
+        # the matlab code.
+        reorg_suppressed = []
+        reorg_unsuppressed = []
+        if nechoes == 1:
```

Comment:
That said, we could do something like this:

```if nechoes == 1 and self.hdr.rhi_user19 > 0:```

This indicates that the edit waveform was no all zeroes.